### PR TITLE
Add support for decimal types in HiveTypeParser

### PR DESCRIPTION
**Pull request by majetideepak**
Add support for decimal types in HiveTypeParser. 
Enable HiveTypeParserTest.
Use VELOX_CHECK, VELOX_FAIL in HiveTypeParser.

### CODE REVIEW

**Comment by majetideepak** [comment]
@Yuhta can you help review this? I am starting to add Decimal Type support for the Parquet Native Reader and want to add incremental changes. This change is required to add a decimal test to the `E2EFilterTest.cpp`. Thanks.

**Comment by Yuhta** [comment]
@majetideepak The tests are failing

File: velox/dwio/dwrf/utils/test/ProtoUtilsTests.cpp [hunk 1]
```diff
@@ -40,17 +40,6 @@ TEST(ProtoUtilsTests, AllTypes) {
   }
 }
 
-TEST(ProtoUtilsTests, UnionDeprecation) {
```
Comment (majetideepak) [inline velox/dwio/dwrf/utils/test/ProtoUtilsTests.cpp:43]: This is moved to `HiveTypeParserTests` since one such test is already present there.

**Comment by majetideepak** [comment]
@Yuhta The tests are now passing. Thanks for the help with the merge.

**Comment by majetideepak** [comment]
@Yuhta any internal test failures due to this PR?

**Comment by Yuhta** [comment]
There are someone catching `std::invalid_argument` explicitly, I can fix it for them though, no worry

**Pull request closed without merging**